servers/src/main/java/tachyon/worker/block/BlockServiceHandler.java

```diff
@@ -87,14 +92,22 @@ public boolean asyncCheckpoint(int fileId) throws TException {
    *
    * @param userId the id of the client requesting the commit
    * @param blockId the id of the block to commit
-   * @throws TException if the block cannot be committed
+   * @throws TException
    */
   // TODO: Reconsider this exception handling
   public void cacheBlock(long userId, long blockId) throws TException {
     try {
       mWorker.commitBlock(userId, blockId);
+    } catch (AlreadyExistsException aee) {
+      throw new TException(aee);
+    } catch (NotFoundException nfe) {
+      throw new TException(nfe);
+    } catch (FailedPreconditionException fpe) {
+      throw new TException(fpe);
     } catch (IOException ioe) {
       throw new TException(ioe);
+    } catch (OutOfSpaceException ooe) {
+      throw new TException(ooe);
```

Comment:
In following PRs, different exceptions will be handled differently in RPC instead of wrapping into `TException`, so it's more informative to list all exceptions here for future refactoring.